compiler/src/org.graalvm.compiler.core.aarch64.test/src/org/graalvm/compiler/core/aarch64/test/AArch64BitFieldTest.java

```diff
@@ -207,7 +207,7 @@ public void testSignedI2LInsert() {
     // SBFIZ with B2I.
     public int signedB2IInsert(int input) {
         byte b = (byte) input;
-        return b << 31;
+        return b << 30;
```

Comment:
Out of curiosity, why did you change this?

Reply:
Investigation took place to determine why and how a negate and shift was generated when there is no apparent negate in signedB2IInsert().

It's quite interesting how we get negate+shift when there is no apparent negate in signedB2IInsert(). The following optimizations result in negate+shift generated for the same operation.

Reassociation phase replaces the left shift (input << 31) with multiply (input * -2147483648). This simplification occurs from tryReplaceWithMulNode() in ReassociatePhase.java Line#163. As 2's complement of 2 << 31 is same as its negative in 4 byte representation, up-down between int and long results in selecting a -ve multiplier. 2. Cannonicalizer phase replaces the multiply with shift+negate in cannonical() from MulNode.java Line#161.

Therefore, as Xiaohong Gong suggested, we simply updated the actual test to use shift with a smaller number. Therefore, we now shift by << 30 instead of 31.